### PR TITLE
refactor: index screen logics separated per resource

### PR DESCRIPTION
**Pull request by donni106**
I now have a version in Code Review where there is a switch between the overview for EventRecords and the others, so that separate files can already be used for EventRecords. This equalises logics on the previously overpowering IndexScreen, which was no longer maintainable.

Logics for event overviews should be handled separately and it should be clearer to understand when something happens. The calendar should now also function as separately as possible with its own query and is no longer dependent on data in the list.

Please check carefully all back and forth on overviews, both with tab navigation and with drawer navigation, jumping around between overviews (especially of the same type as "Nachrichten" and "Meldungen"). Filters and data must always be correct.

SVA-1114

### CODE REVIEW

**Review by ardasnturk — COMMENTED** [review]
The simplicity and clarity of the IndexScreen has been nicely enhanced. But Overviews and EventRecords are still confusing, although not as much as the previous IndexScreen. I have detected the following error. 

First I am in IndexScreen in calendar view and I go to DetailScreen of an event, then I do pullToRefresh and new data is shown. Then I went back to the IndexScreen and saw that the SectionList was not showing the day I selected from the calendar. I don't know where this is coming from. 

Also, on the Volunteer screens where we use the Calendar, the application crashes. But as far as I have noticed, the application crashes on many pages for Volunteer in general, not only on the pages where we use the calendar. Maybe it's due to lack of data, but I have no idea.

|IndexScreen|DetailScreen|DetailScreen (pullToRefresh)|IndexScreen|
|--|--|--|--|
Simulator Screenshot - iPhone 14 Pro Max - 2023-10-05 at 15 13 33|Simulator Screenshot - iPhone 14 Pro Max - 2023-10-05 at 15 13 35|Simulator Screenshot - iPhone 14 Pro Max - 2023-10-05 at 15 16 25|Simulator Screenshot - iPhone 14 Pro Max - 2023-10-05 at 15 13 25

File: src/components/Calendar.tsx [hunk 1]
```diff
@@ -32,137 +34,160 @@ const DayComponent = (props: BasicDayProps & { date?: DateData }) => (
...
+
+    // highlight selected day
+    dates[selectedDay] = {
+      ...(dates[selectedDay] ?? {}),
+      selected: true,
+      selectedColor: colors.lighterPrimary
+    };
+
+    return dates;
+  }, [query, data, selectedDay, dotCount]);
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:143]: no need to call an external method and pass data that is already existing in the component and same `useMemo` way as `listItems`

File: src/components/Calendar.tsx [hunk 2]
```diff
@@ -32,137 +34,160 @@ const DayComponent = (props: BasicDayProps & { date?: DateData }) => (
 );
 
 type Props = {
+  isListRefreshing: boolean;
   query: string;
   queryVariables?: { dateRange?: string[]; contentContainerId?: number };
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:39]: thanks, need to check if other components using the calendar also needs this.
it signals the calendar, that the main list screen is pulled to refresh. knowing this, the calendar can refresh itself.

File: src/screens/WasteCollectionScreen.js [hunk 3]
```diff
@@ -36,42 +45,12 @@ import { getInAppPermission, showPermissionRequiredAlert } from '../pushNotifica
...
-  // highlight today
-  markedDates[today] = {
-    ...(markedDates[today] ?? {}),
-    selected: true,
-    selectedColor: colors.lighterPrimary
-  };
-
-  return markedDates;
-};
+const today = moment().format('YYYY-MM-DD');
```
Comment (ardasnturk) [inline src/screens/WasteCollectionScreen.js:53]: We use the same code in many places. Why don't we create a helper for that?

File: src/components/Calendar.tsx [hunk 4]
```diff
@@ -29,13 +29,20 @@ const { DOT_SIZE, MAX_DOTS_PER_DAY } = CALENDAR;
...
 const today = moment().format('YYYY-MM-DD');
 
-export const Calendar = ({ isListRefreshing, query, queryVariables, navigation }: Props) => {
+export const Calendar = ({
+  isListRefreshing,
+  query,
+  queryVariables,
+  navigation,
+  additionalData
+}: Props) => {
```
Comment (ardasnturk) [inline src/components/Calendar.tsx:45]: Wouldn't alphabetical sorting be better?

File: src/components/Calendar.tsx [hunk 5]
```diff
@@ -29,13 +29,20 @@ const { DOT_SIZE, MAX_DOTS_PER_DAY } = CALENDAR;
 type Props = {
   isListRefreshing: boolean;
   query: string;
-  queryVariables?: { dateRange?: string[]; contentContainerId?: number };
+  queryVariables: { dateRange?: string[]; contentContainerId?: number };
   navigation: StackNavigationProp<any>;
+  additionalData?: any;
 };
```
Comment (ardasnturk) [inline src/components/Calendar.tsx:35]: I see that the alphabetical sorting is also missing here.

File: src/components/screens/EventRecords.js [hunk 6]
```diff
@@ -200,15 +191,15 @@ export const EventRecords = ({ navigation, route }) => {
       showVolunteerEvents && refetchVolunteerEvents();
     }
     setRefreshing(false);
-  }, [isConnected, setRefreshing, showVolunteerEvents, refetch, refetchVolunteerEvents]);
+  }, [isConnected, setRefreshing, refetch, showVolunteerEvents, refetchVolunteerEvents]);
```
Comment (ardasnturk) [inline src/components/screens/EventRecords.js:194]: Why did you need to change the sorting?

File: src/components/volunteer/VolunteerHomeSection.tsx [hunk 7]
```diff
@@ -85,14 +89,16 @@ export const VolunteerHomeSection = ({
...
         {showCalendar ? (
           <Calendar
+            isListRefreshing={isLoading}
             query={query}
             queryVariables={queryVariables}
-            calendarData={sectionData}
-            isLoading={isLoading}
             navigation={navigation}
+            additionalData={sectionData}
           />
```
Comment (ardasnturk) [inline src/components/volunteer/VolunteerHomeSection.tsx:102]: I see there is no alphabetical sorting here either.

File: src/screens/volunteer/VolunteerIndexScreen.tsx [hunk 8]
```diff
@@ -106,11 +109,11 @@ export const VolunteerIndexScreen = ({ navigation, route }: StackScreenProps<any
...
             showCalendar ? (
               <Calendar
+                isListRefreshing={isLoading}
                 query={query}
                 queryVariables={queryVariables}
-                calendarData={data}
-                isLoading={isLoading}
                 navigation={navigation}
+                additionalData={data}
               />
```
Comment (ardasnturk) [inline src/screens/volunteer/VolunteerIndexScreen.tsx:117]: I see there is no alphabetical sorting here either.

File: src/components/Calendar.tsx [hunk 9]
```diff
@@ -29,13 +29,20 @@ const { DOT_SIZE, MAX_DOTS_PER_DAY } = CALENDAR;
...
 
 const today = moment().format('YYYY-MM-DD');
 
-export const Calendar = ({ isListRefreshing, query, queryVariables, navigation }: Props) => {
+export const Calendar = ({
+  isListRefreshing,
+  query,
+  queryVariables,
+  navigation,
+  additionalData
```
Comment (ardasnturk) [inline src/components/Calendar.tsx:44]: Shouldn't `additionalData` be included in the `markedDates` function? At the moment I can't see the dots in Volunteer's calendar, and I can't click on any day in the calendar because there are no dots.

**Review by ardasnturk — CHANGES_REQUESTED** [review]

File: src/screens/volunteer/VolunteerIndexScreen.tsx (hunk 8 above)
Comment (ardasnturk) [reply src/screens/volunteer/VolunteerIndexScreen.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/940/commits/f653378eefcca79cef1def80dfa038e995b1bccf

File: src/components/volunteer/VolunteerHomeSection.tsx (hunk 7 above)
Comment (ardasnturk) [reply src/components/volunteer/VolunteerHomeSection.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/940/commits/f653378eefcca79cef1def80dfa038e995b1bccf

File: src/components/screens/EventRecords.js (hunk 6 above)
Comment (ardasnturk) [reply src/components/screens/EventRecords.js]: https://github.com/smart-village-solutions/smart-village-app-app/pull/940/commits/f653378eefcca79cef1def80dfa038e995b1bccf

File: src/components/Calendar.tsx (hunk 5 above)
Comment (ardasnturk) [reply src/components/Calendar.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/940/commits/f653378eefcca79cef1def80dfa038e995b1bccf

File: src/components/Calendar.tsx (hunk 4 above)
Comment (ardasnturk) [reply src/components/Calendar.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/940/commits/f653378eefcca79cef1def80dfa038e995b1bccf

File: src/screens/WasteCollectionScreen.js (hunk 3 above)
Comment (donni106) [reply src/screens/WasteCollectionScreen.js:53]: haven't found the time to think about it and find a good refactoring.

File: src/components/Calendar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx]: should be done with updated commits

File: src/components/Calendar.tsx (hunk 9 above)
Comment (donni106) [reply src/components/Calendar.tsx]: f602ddd

image

**Review by ardasnturk — APPROVED** [review]
